### ISSUE TITLE
Swan DevGrant<go-computing-provider>

### ISSUE THREAD

**Issue by Normalnoise** [swanchain/devgrants#21]
# Open Grant Proposal: `go-computing-provider`

**Project Name:** `go-computing-provider`

**Proposal Category:** `Applications`. 

**Individual or Entity Name:** `FogMeta Team`

**Proposer:** `Normalnoise`

**(Optional) Swan ecosystem affiliations:** `It will register to Orchestrator in Swan network, all conputing task can be assigned to the computing provider` 

**(Optional) Technical Sponsor:** `flyworker`

**Do you agree to open source all work you do on behalf of this RFP under the MIT/Apache-2 dual-license?:** `YES`.

# Project Summary

`go-computing-provider`is an individual or organization that participates in the decentralized computing network by offering computational resources such as processing power (CPU and GPU), memory, storage, and bandwidth. Their primary role is to execute tasks assigned by [Orchestrator](https://orchestrator.swanchain.io/) on the [Swan chain](https://swanchain.io/).

## Impact 

All resource providers can lease their conputing resource by installing the `go-computing-provider` software. So far this is the only computing software available on swan chain.

## Outcomes

 - `Computing-provider` code repository
 - the documentation incluede build & config & installation
 - upgrade guideline for every time

## Total Budget Requested

| Milestone # | Description                                                  | Deliverables                                                 | Completion Date | Funding |
| ----------- | ------------------------------------------------------------ | ------------------------------------------------------------ | --------------- | ------- |
| Milestone1  | Design and implement the CP mechanism based on the  Orchestrator on Swanchain. | go-computing-provider                          | 4 Weeks             | 50,000  SWAN |
| Milestone2  | Implement computing provider for receiving the space task, and provide the installation documentation | go-computing-provider & DOC            | 3 Weeks            | 100,000  SWAN |
| Milestone3  | Add the task verification mechanism, build and container log WSS, support UBI task. | support the space  | 3 Week             | 50,000 SWAN |

## Maintenance and Upgrade Plans

**Maintenance**: We will conduct routine maintenance activities to ensure the stability and reliability of the CP project. This includes code reviews, bug fixes, and performance optimizations.

**Community Engagement**: We aim to foster community engagement by encouraging open-source contributions, feedback, and collaboration. This will help us identify areas for improvement and drive innovation.

**Version Control and Release Management**: We will utilize Github to implement a structured release management process to manage changes effectively and ensure seamless deployments.

# Team

## Team Members

Leo: Director of Engineering 

Will: Software Engineer

Sonic: Software Engineer

Eric: Operations Engineer

Kira: Marketing Manager

## Team Member Github Profiles

Leo: [GitHub Profiles](https://github.com/Normalnoise)

Will: [GitHub Profiles](https://github.com/will2z)

Sonic: [GitHub Profiles](https://github.com/sonic-chain) 

Eric: [GitHub Profiles](https://github.com/Eric-3A)

Kira: [GitHub Profiles](https://github.com/Kirameta)

## Relevant Experience

Our team has extensive experience building blockchain-based applications and decentralized infrastructure, like IPFS, Filecoin, and ETH ecosystem .

Leo, our Director of Engineering, has over 4 years of experience leading engineering teams to successfully ship complex software products, and experience in in-depth research and participation in the development and design of multiple blockchain projects. 

Will and Sonic are full-stack engineers with special expertise in Kubenates, Cloud platform, smart contract development and blockchain integrations. Together, they have contributed to several prominent protocols.

Eric has expertise in IT operations and devops with over 8 years of IT operation and maintenance experience in large-scale Internet financial technology industry. He will ensure our infrastructure is scalable, secure, and resilient. 

Kira leads marketing and business development with with over 5 years of experience launching blockchain projects and growing communities.

With deep blockchain engineering talent and experience launching successful crypto projects, we believe our team has the ideal background to execute on the vision of this grant. We are excited at the prospect of receiving a Swan Grant to turn our ideas into reality.

## Team code repositories

https://github.com/swanchain/go-computing-provider

# Additional Information

contact@fogmeta.com

**Comment by JL289** [comment]
Hello there @Normalnoise,

We'd like to know more about the project, could you please provide more information pertaining to: 

- Marketing and Distribution plans for your application? No plan was provided. 

Sorry for the wait, let us know if you have any questions! 

Cheers,

**Comment by Normalnoise** [comment]
yes, @JL289 `go-computing-provider` has served several swan testnet, including `Mars-testnet`, `Jupyter testnet`,`Proxima-testnet`. Until to now, It is still the only available CP client. In the current Proxima testnet, there are about 1800 CP running it. 

In the future, we will continue to support Swan Testnet and Swan Mainnet. the issues from community will be supported

**Comment by JL289** [comment]
Hey there @Normalnoise, 

Thank you for your answer, we will process the information and get back to you shortly. 

Cheers,

**Comment by CryptoSenorita** [comment]
Hi go-computing-provider team,

Thank you for your patience with our evaluation process. Please share your proof of integration with Swan Chain (in videos or pictures) here, along with the progress, results, and work demos. We will inform you of the next steps once the review is successfully completed. Thank you.

Swan Chain Team

**Comment by Normalnoise** [comment]
@CryptoSenorita  thanks for your review. `go-computing-provider` has completed V1, which has been used by swan chain's CPs.  And the project repo has been submitted to swanchain Github: https://github.com/swanchain/go-computing-provider.

 - All the CPs are using this client, you can see the CP list here: https://provider.swanchain.io/rankings/fcp
<img width="1192" alt="image" src="https://github.com/user-attachments/assets/7cd4362f-05ec-4c26-b909-af8c0c01bc66">

 - As a FCP, `go-computing-provider` can accept the AI task from Orchestrator on Swanchain 
 
  - As an ECP, `go-computing-provider` can complete the zk task from ZK Engine on Swanchain
  
  - Completed the task verification mechanism, support build log and container log(WSS), and support UBI task.
  
   - Has released 22 versions and solved 33 issues.

We utilise GitHub to implement a structured release management process to manage changes effectively and ensure seamless deployments. Actively help community users to solve deployment problems, environment configuration problems, repair mechanism design, and other problems; after several rounds of testing, complete a relatively complete version (V1.0.0).

**Comment by CryptoSenorita** [comment]
Hi go-computing-provider team,

Thank you for your patience in our evaluation process. We have approved your proposal. We will email you back to discuss the next step.

Swan Chain team
